internal/server/middlewares.go

```diff
@@ -144,10 +150,8 @@ func requestLogger(lg *zap.Logger) middleware {
 			case http.MethodGet:
 				break
 			default:
-				buf, err := io.ReadAll(req.Body)
-				if err != nil {
-					lg.Debug("error reading request body: %v", zap.String("error", err.Error()))
-				} else if len(buf) > 0 {
+				if noOfBytes > 0 {
+					buf, _ := io.ReadAll(bodyCopy)
```

Comment:
we should be able to directly use `bodyCopy` instead of reading it again to `buf`.

additionally, let's also try to make a habit to handle error and not skipping it.
and add context to the error (e.g. wrapping it if returning and add message when logging it)